airflow-core/docs/howto/deadline-alerts.rst

```diff
@@ -0,0 +1,263 @@
+
+Airflow provides several built-in reference points that you can use with DeadlineAlert:
+
+``DeadlineReference.DAGRUN_QUEUED_AT``
+    Measures time from when the DagRun was queued. Useful for monitoring resource constraints.
+
+``DeadlineReference.DAGRUN_LOGICAL_DATE``
+    References when the DAG run was scheduled to start. For example, setting an interval of
+    ``timedelta(minutes=15)`` would trigger the alert if the DAG hasn't completed 15 minutes
+    after it was scheduled to start, regardless of when (or if) it actually began executing.
+    Useful for ensuring scheduled DAGs complete before their next scheduled run.
+
+``DeadlineReference.FIXED_DATETIME``
+    Specifies a fixed point in time. Useful when DAGs must complete by a specific time.
+
+Here's an example using a fixed datetime:
+
+.. code-block:: python
+
+    tomorrow_at_ten = datetime.combine(datetime.now().date() + timedelta(days=1), time(10, 0))
+
+    with DAG(
+        dag_id="fixed_deadline_alert",
+        deadline=DeadlineAlert(
+            reference=DeadlineReference.FIXED_DATETIME(tomorrow_at_ten),
+            interval=timedelta(minutes=-30),  # Alert 30 minutes before the reference.
+            callback=SmtpNotifier(
+                to="team@example.com",
+                subject="Report will be late",
+                html_content="The report will not be ready 30 minutes before the deadline.",
+            ),
+        ),
+    ):
+        EmptyOperator(task_id="example_task")
+
+The timeline for this example would look like this:
+
+::
+
+    |------|----------|---------|------------|--------|
+         Queued     Start    Deadline    Reference
+         09:15      09:17     09:30       10:00
+
+.. note::
+    Note that since the interval is a negative value, the deadline is before the reference in this case.
+
+Using Callbacks
+---------------
+
+When a deadline is exceeded, the callback is executed. You can use any async :doc:`Notifier </howto/notifications>`
+or create a custom callback function.
+
+Using Built-in Notifiers
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here's an example using the Slack notifier if the DagRun has not finished within 30 minutes of it being queued:
+
+.. code-block:: python
+
+    with DAG(
+        dag_id="slack_deadline_alert",
+        deadline=DeadlineAlert(
+            reference=DeadlineReference.DAGRUN_QUEUED_AT,
+            interval=timedelta(minutes=30),
+            callback=SlackNotifier(
+                slack_conn_id="slack_default",
+                channel="#alerts",
+                text="DAG {{ dag.dag_id }} has been running for more than 30 minutes since being queued.",
+                username="Airflow Alerts",
+            ),
+        ),
+    ):
+        EmptyOperator(task_id="example_task")
+
+Creating Custom Callbacks
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can create custom callbacks for more complex handling. The ``callback_kwargs`` specified in
+the ``DeadlineAlert`` are passed to the callback function.
```

Comment:
This and the code snippets need updating after https://github.com/apache/airflow/pull/53951#pullrequestreview-3080641355 right?

Reply:
That is possible.   I'll keep an eye on the in-flight PRs and adjust accordingly.